fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -67,21 +68,31 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         }
 
         override fun onLoggedOut() {
-            viewLifecycleOwner.lifecycleScope.launch {
-                findNavController().popBackStack()
+            forceCloseAndClearData()
+        }
+    }
 
-                // Remove the device name when we log out.
-                context?.let {
-                    val deviceNameKey = it.getPreferenceKey(R.string.pref_key_sync_device_name)
-                    preferenceManager.sharedPreferences?.edit()?.remove(deviceNameKey)?.apply()
-                }
+    private fun forceCloseAndClearData() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            findNavController().popBackStack()
+
+            // Remove the device name when we log out.
+            context?.let {
+                val deviceNameKey = it.getPreferenceKey(R.string.pref_key_sync_device_name)
+                preferenceManager.sharedPreferences?.edit()?.remove(deviceNameKey)?.apply()
             }
         }
     }
 
     override fun onResume() {
         super.onResume()
         showToolbar(getString(R.string.preferences_account_settings))
+
+        // The user might have deleted the account through the "manage account" tab, in which case
+        // the account settings fragment should not be shown.
+        if (accountManager.accountProfile() == null) {
```

Comment:
This is race-y. Please use an account observer or observe the `SyncStore`.

Reply:
I don't think it is, the CustomTabActivity is closed only when the accountManager moves into loggedOut state, so there is no change the profile exist at this point.

Also, account observer in this fragment wouldn't catch the delete event, because the callback is called when the user is on CustomTabActivity

Reply:
My thinking was, we [shouldn't](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt#62) be showing the fragment if the user is logged out or there is an Auth problem since it's closed automatically. But the fragment isn't active when the "accountDeleted" is called, so I inroduced a check during fragment creation. Isn't profile always available is the user in logged in state and there is no auth problem? We probable could do some other check to determine if the user is logged it.

Reply:
> Isn't profile always available is the user in logged in state and there is no auth problem?

No, it isn't, which is why you need to always observe for when it's available.